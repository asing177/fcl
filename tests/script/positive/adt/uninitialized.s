type Direction = Left | Right;

global int i;

@initial
start(Direction dir) {
  case(dir) {
    Left -> { i = 10; transitionTo(:assigned); };
    Right -> transitionTo(:unassigned);
  };
}

@assigned
increase() {
  i = i + 1;
  transitionTo(:assigned);
}

@assigned
assignedStop() {
  terminate();
}

@unassigned
assign(Direction dir) {
  case(dir) {
    Left -> { i = 10; transitionTo(:assigned); };
    Right -> transitionTo(:unassigned);
  };
}
