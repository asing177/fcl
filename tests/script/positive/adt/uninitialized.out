 
Method arguments and effects:
start: (direction) -> {write i}
increase: () -> {read i, write i}
assignedStop: () -> {write via terminate}
assign: (direction) -> {write i}